plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -2057,3 +2097,433 @@ describe('PersonState.update()', () => {
+        const distinctIds = await fetchDistinctIds()
+        expect(distinctIds).toEqual([
+            ['first', persons[0].id],
+            ['second', persons[0].id],
+        ])
+
+        // verify Postgres person_id overrides
+        const overrides = await fetchPersonIdOverrides()
+        expect(overrides).toEqual([[second.uuid, first.uuid]])
+
+        // verify running merge again doesn't change the state
+        await updatePersonStateFromEvent({
+            event: '$identify',
+            distinct_id: 'first',
+            properties: {
+                $anon_distinct_id: 'second',
+            },
+        })
+    })
+
+    it('handles a chain of overrides being applied concurrently', async () => {
+        await updatePersonStateFromEvent({
+            event: 'event',
+            distinct_id: 'first',
+        })
+
+        await updatePersonStateFromEvent({
+            event: 'event',
+            distinct_id: 'second',
+        })
+
+        await updatePersonStateFromEvent({
+            event: 'event',
+            distinct_id: 'third',
+        })
+
+        const [first, second, third] = await fetchPostgresPersons()
+
+        // We want to simulate a concurrent update to person_overrides. We do
+        // this by first mocking the implementation to block at a certain point
+        // in the transaction, then running the updatePersonStateFromEvent
+        // function twice. We then wait for them to block before letting them
+        // resume.
+        let resumeExecution: (value: unknown) => void
+
+        const postgresTransaction = hub.db.postgresTransaction.bind(hub.db)
+        jest.spyOn(hub.db, 'postgresTransaction').mockImplementation(async (tag: string, transaction: any) => {
+            if (tag === 'mergePeople') {
+                return await postgresTransaction(tag, async (client) => {
+                    if (resumeExecution) {
+                        resumeExecution(undefined)
+                    } else {
+                        await new Promise((resolve) => {
+                            resumeExecution = resolve
+                        })
+                    }
+
+                    return await transaction(client)
+                })
+            } else {
+                return await postgresTransaction(tag, transaction)
+            }
+        })
+
+        await expect(
+            Promise.all([
+                updatePersonStateFromEvent({
+                    event: '$identify',
+                    distinct_id: 'second',
+                    properties: {
+                        $anon_distinct_id: 'first',
+                    },
+                }),
+                updatePersonStateFromEvent({
+                    event: '$identify',
+                    distinct_id: 'third',
+                    properties: {
+                        $anon_distinct_id: 'second',
+                    },
+                }),
+            ])
+        ).rejects.toThrow()
```

Comment:
This test is failing, but I don't really understand how the whole `postgresTransaction` mock works or is intended to work. This is not throwing as both queries are running sequentially. 

I'm also confused why we `await` the same below but without expecting it to `throw`. 

:confused: